flair_text.py

```diff
@@ -0,0 +1,105 @@
+__copyright__ = "Copyright (c) 2021 Jina AI Limited. All rights reserved."
+__license__ = "Apache-2.0"
+
+from typing import Union, Tuple, List, Any, Iterable
+
+import numpy as np
+import torch
+from jina import Executor, requests, DocumentArray
+
+
+def _batch_generator(data: List[Any], batch_size: int):
+    for i in range(0, len(data), batch_size):
+        yield data[i:min(i + batch_size, len(data))]
```

Comment:
As mentioned to florian, no need to write `min`. you can set this as `data[i: i + batch_size]` the `min` is done at `C` level faster and much more efficient.